clang/test/Driver/sycl-offload.cpp

```diff
@@ -170,3 +170,11 @@
 /// Check if the clang with fsycl adds C++ libraries to the link line
 //  RUN:  %clang -### -target x86_64-unknown-linux-gnu -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK-FSYCL-WITH-CLANG %s
 // CHECK-FSYCL-WITH-CLANG: "-lstdc++"
+
+/// Check selective passing of -emit-only-kernels-as-entry-points to sycl-post-link tool
+// RUN: %clang -### -target spir64_fpga -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK_SYCL_POST_LINK_OPT_PASS %s
+// RUN: %clang -### -target spir64_gen -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK_SYCL_POST_LINK_OPT_PASS %s
+// CHECK_SYCL_POST_LINK_OPT_PASS: sycl-post-link{{.*}}emit-only-kernels-as-entry-points
+// RUN: %clang -### -target spir64_gen -fsycl-emit-external-funcs-as-entry-points -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK_SYCL_POST_LINK_OPT_NO_PASS %s
```

Comment:
Should these be using `--target=x86_64-unknown-linux-gnu -fsycl-targets=spir64_[fpga|gen]`?

Reply:
Yes. It should. I am not sure where I got this from. Addressed in next commit. Thanks